massgen/tests/test_checkpoint_mcp_standalone.py

```
@@ -903,6 +1010,386 @@ def test_opening_uses_broadened_framing(self):
+
+class TestSingleCheckpointMode:
+    """Single-checkpoint mode strips the recheckpoint affordance from every
+    model-visible surface: reviewer prompt (terminal docs, examples, depth
+    directive), VALID_TERMINALS, and the executor-facing instructions template.
+
+    Runtime backstop exists but is defense-in-depth only — the primary
+    mechanism is source-removal from prompts.
+    """
+
+    _DEFAULT_TASK = "Test user task: do the requested thing."
+    _DEFAULT_ENV: dict = {}
```

Ruff (0.15.10)

[warning] Mutable default value for class attribute

(RUF012)